src/types/Post.ts

```diff
@@ -0,0 +1,11 @@
+import { Comment } from './Comment';
+import { User } from './User';
+
+export interface Post {
+  userId: number,
+  id: number,
+  title: string,
+  body: string,
+  user: User | undefined,
+  comments: Comment[] | undefined,
```

Comment:
comments will not be undefined. Filter will return an empty array
```suggestion
  comments: Comment[],
```